lib/vagrant-docker-compose/config.rb

```diff
@@ -1,12 +1,30 @@
 module VagrantPlugins
   module DockerComposeProvisioner
     class Config < Vagrant.plugin("2", :config)
-      attr_accessor :yml, :rebuild
+      attr_accessor :yml, :rebuild, :executable
 
-      def yml=(yml)
-        raise DockerComposeError, :yml_must_be_absolute if !Pathname.new(yml).absolute?
-        @yml = yml
+      def initialize
+        @executable = UNSET_VALUE
+        @yml = UNSET_VALUE
       end
+
+      def finalize!
+        @yml = nil if @yml == UNSET_VALUE
+        @executable = '/usr/local/bin/docker-compose' if @executable == UNSET_VALUE
+      end
+
+      def validate(machine)
+        errors = []
+
+        if !@yml
+          errors << I18n.t(:yml_not_set, scope: [:errors])
+        elsif !Pathname.new(@yml).absolute?
+          errors << I18n.t(:yml_must_be_absolute, scope: [:errors])
+        end
+
+        { "docker-compose provider" => errors }
```

Comment:
i'm not familiar with how the error mapping here works. because this is a provisioner plugin not a provider plugin, should this be `docker-compose provisioner`? does the key matter or what impact does it have?